Update to Go testing page - adding debugging info (#468)
temporalio/documentation: docs/go/testing.md

---
id: testing

> title: Testing and Debugging
> sidebar_label: Testing and Debugging

Reviewer: ```suggestion
title: Testing and debugging
sidebar_label: Testing and debugging
```

Sentence casing

	s.Equal(value, 100)
}
```

## Debugging

> In addition to writing unit and integration tests, debugging your Workflows is also a very valuable testing tool.
> You can debug your Workflow code using a debugger provided by your favorite IDE.

Reviewer: ```suggestion
You can use a debugger tool provided by your favorite IDE to debug your Workflow Definitions prior to testing or executing them.
```

> Note that when debugging your Workflow code, the Temporal Go SDK includes deadlock detection which fails a Workflow Task in case the code blocks over a second without relinquishing execution control.

Reviewer: A couple of things here:
1. Instead of "Workflow code" lets try to use Workflow Definition when referring to static Workflow code.
2. A Workflow Task is something very specific and I don't think an IDE debugging tool knows anything about a Workflow Task - it sounds like we are referring to a Workflow "awaitable" ? - like an Activity or something? 

```suggestion
The Temporal Go SDK includes deadlock detection which fails a Workflow awaitable (Activity Execution, Signal, Query) in case the code blocks over a second without relinquishing execution control.
```

Author: idk what is am "awaitable" . the code docs say "workflow task"

> Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow code during debugging.

Reviewer: ```suggestion
Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow Definitions during debugging.
```

> To alleviate this issue, you can set the TEMPORAL_DEBUG environment variable to `true` before debugging your Workflow code.

Reviewer: ```suggestion
To alleviate this issue, you can set the TEMPORAL_DEBUG environment variable to `true` before debugging your Workflow Definition.
```

> Make sure to set TEMPORAL_DEBUG to true only during debugging.

Reviewer: ```suggestion
::: note

Make sure to set TEMPORAL_DEBUG to true only during debugging.

:::
```

	s.Equal(value, 100)
}
```

## Debugging

You can use a debugger tool provided by your favorite IDE to debug your Workflow Definitions prior to testing or executing them.

> Note that when debugging your Workflow code, the Temporal Go SDK includes deadlock detection which fails a Workflow Task in case the code blocks over a second without relinquishing execution control.

Reviewer: OK.
```suggestion
The Temporal Go SDK includes deadlock detection which fails a Workflow Task in case the code blocks over a second without relinquishing execution control.
```

Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow Definitions during debugging.

> To alleviate this issue, you can set the TEMPORAL_DEBUG environment variable to `true` before debugging your Workflow Definition.
> ::: note

Reviewer: ```suggestion

To alleviate this issue, you can set the TEMPORAL_DEBUG environment variable to `true` before debugging your Workflow Definition.

::: note
```

	s.Equal(value, 100)
}
```

## Debugging

You can use a debugger tool provided by your favorite IDE to debug your Workflow Definitions prior to testing or executing them.

The Temporal Go SDK includes deadlock detection which fails a Workflow Task in case the code blocks over a second without relinquishing execution control.
Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow Definitions during debugging.

> 

Reviewer: ```suggestion
```
